src/Containers/MenuItemCard/MenuItemCard.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Main, MainInterface, Responsive, ResponsiveInterface } from '@Utils/BaseStyles';
+import { transition } from '@Utils/Mixins';
+// eslint-disable-next-line import/no-cycle
+import { LoyaltyPoints, LimitedTimeBanner, SaleTag } from '../../index';
+
+export interface MenuItemCardProps
+    extends MainInterface, ResponsiveInterface, React.HTMLAttributes<HTMLDivElement> {
+    LoyaltyPoints?: React.ReactElement;  
+    LimitedTimeBanner?: React.ReactElement;  
+    SaleTag?: React.ReactElement;  
+    itemImage: string, // Image for the Menu Item Card, hosted on any server and converted to 350px x 200px 
+    itemName: string, // Item name, heightFitContent adjusts the item card size to fit any text (keeping min size of 350px)
+    itemPrice: number, // Price of the item, will be adjusted by saleAmount and will reduce + round if above 1000
+    itemPriceLimit: number, // Limit before item price is divided by 1000, should be greater than 1000
+    animated?: boolean;
+    flat?: boolean;
+    widthFitContent?: boolean;
+    heightFitContent?: boolean; 
+    sale?: boolean;
+    soldOut?: boolean;
+    loyaltyamount: number,
+    loyaltypointlimit: number,
+    saleAmount: number,
+    minsRemaining: number,   
+}
+
+export const ItemCard: React.FC<MenuItemCardProps> = ({
+    children, 
+    ...props
+}): React.ReactElement => <MenuItemCardBox {...props}>{children}</MenuItemCardBox>;
+
+export const MenuItemCard: React.FC<MenuItemCardProps> = ({
+    animated,
+    flat,
+    widthFitContent,
+    heightFitContent,
+    sale,
+    soldOut,
+    itemImage, 
+    itemName,
+    itemPrice,
+    itemPriceLimit,
+    saleAmount,
+    loyaltyamount,
+    loyaltypointlimit, 
+    minsRemaining, 
+    ...props
+
+}): React.ReactElement => (  
+    <ItemCard {...props} itemImage={itemImage} itemName={itemName} itemPrice={itemPrice} animated={animated} 
+        flat={flat} widthFitContent={widthFitContent} sale={sale} soldOut={soldOut} saleAmount = {saleAmount} itemPriceLimit = {itemPriceLimit}
+        loyaltyamount = {loyaltyamount} minsRemaining = {minsRemaining} loyaltypointlimit = {loyaltypointlimit} heightFitContent = {heightFitContent}> 
+
+        <img src={itemImage} alt={itemName} style={{ width: 350, height: 200 }}/>
+        <ItemHeader>{itemName}</ItemHeader>
+        { soldOut && <SoldOutBox>Sold Out</SoldOutBox>}
+
+        { !!minsRemaining && 
+        <LimitedTimeBannerPosition> <LimitedTimeBanner minsRemaining={minsRemaining}/></LimitedTimeBannerPosition> } 
+    
+        { !!loyaltyamount && 
+        <LoyaltyPointsPosition> <LoyaltyPoints loyaltyamount={loyaltyamount} loyaltypointlimit = {loyaltypointlimit} /></LoyaltyPointsPosition> }
+
+        { !!saleAmount && 
+        <SaleTagPosition><SaleTag saleAmount={saleAmount}/></SaleTagPosition> }
```

Comment:
Can you abstract this all out into a getMenuItemStatus function.

What happens if someone passes in two conflicting states. We will need to override.